tests/Unit/IMAP/Charset/ConverterTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright 2023 Anna Larch <anna.larch@gmx.net>
+ * @author Anna Larch <anna.larch@gmx.net>
+ *
+ * Mail
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Tests\Unit\IMAP\Charset;
+
+use ChristophWurst\Nextcloud\Testing\TestCase;
+use Horde_Mime_Part;
+use OCA\Mail\IMAP\Charset\Converter;
+use function mb_convert_encoding;
+
+class ConverterTest extends TestCase {
+
+	public Converter $converter;
+	protected function setUp(): void {
+		parent::setUp();
+
+		$this->converter = new Converter();
+	}
+
+	/**
+	 * @dataProvider dataProviderMimeParts
+	 */
+	public function testConvert($mimePart, $expected): void {
+		$actual = $this->converter->convert($mimePart);
+		$this->assertEquals($expected, $actual);
+		$isUtf8 = mb_check_encoding($actual, 'UTF-8');
+		$this->assertTrue($isUtf8);
+	}
+
+	public function dataProviderMimeParts(): array {
+		// UTF8
+		$utfMimePart = new Horde_Mime_Part();
+		$utfMimePart->setType('text/plain');
+		$utfMimePart->setCharset('UTF-8');
+		$utfMimePart->setContents('😊');
+		// Hebrew
+		$iso88591MimePart = new Horde_Mime_Part();
+		$iso88591MimePart->setType('text/plain');
+		$iso88591MimePart->setCharset('ISO-8859-1');
+		$iso88591MimePart->setContents(mb_convert_encoding('Ümlaut', 'ISO-8859-1', 'UTF-8'));
+		$iso88591MimePart_noCharset = new Horde_Mime_Part();
+		$iso88591MimePart_noCharset->setContents('בה בדף לחבר ממונרכיה, בקר בגרסה ואמנות דת');
+		// Japanese
+		$iso2022jpMimePart = new Horde_Mime_Part();
+		$iso2022jpMimePart->setType('text/plain');
+		$iso2022jpMimePart->setCharset('ISO-2022-JP');
+		$iso2022jpMimePart->setContents(mb_convert_encoding('外せ園査リツハワ題', 'ISO-2022-JP', 'UTF-8'));
+		$iso2022jpMimePart_noCharset = new Horde_Mime_Part();
+		$iso2022jpMimePart_noCharset->setContents('外せ園査リツハワ題');
+		// Korean - not in mb nor iconv
+		// $iso106461MimePart = new Horde_Mime_Part();
```

Comment:
ISO 10646-1 is impossible to get working for me. The coding is not mentioned in any report. Can we skip this?